### PR TITLE
Fix broken link in Dynamo's guards doc (#102183)

### PR DESCRIPTION
**Pull request by weimingzha0**
This PR fixes broken link for the code referenced in the guards doc.

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: docs/source/compile/guards-overview.rst
Comment (awgu) [inline docs/source/compile/guards-overview.rst:100]: It might be more sustainable to switch this with a permalink?

Comment (weimingzha0) [reply docs/source/compile/guards-overview.rst:100]: Agree

**Comment by mikaylagawarecki** [comment]
Hey @weimingzha0 could you sign the [cla agreement in this comment](https://github.com/pytorch/pytorch/pull/102185#issuecomment-1561731531) please

**Review by mikaylagawarecki — APPROVED** [review]

**Comment by weimingzha0** [comment]
> Hey @weimingzha0 could you sign the [cla agreement in this comment](https://github.com/pytorch/pytorch/pull/102185#issuecomment-1561731531) please

Sure. Waiting for approval from our legal team

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**
